widgets/pulseaudio.lua

```diff
@@ -33,44 +33,49 @@ local function factory(args)
     --  '/device\\.string/ p' prints the line containing 'device.string:'.
     
     local settings    = args.settings or function()
+            widgettext  = volume_now.left .. "%"
+            tooltiptext = volume_now.left .. "% (" .. devicetype .. " " .. volume_now.index .. " \"" .. volume_now.device .. "\")"
             if volume_now.muted == "yes" then
-                widget:set_text("Mute")
-            else
-                widget:set_text(volume_now.left .. "%")
+                widgettext  = "Mute"
+                tooltiptext = "Mute/" .. tooltiptext
             end
+            widget:set_text(widgettext)
+            tooltip:set_text(tooltiptext)
         end
-        
+    
     pulseaudio.widget = wibox.widget.textbox()
-
+    pulseaudio.tooltip = awful.tooltip({ objects = { pulseaudio.widget } })
```

Comment:
I meant that it would be difficult to define the tooltip in the `rc.lua` and then keep it's contents synced to the widget. On the contrary by defining it right here it is easily accessible from the `settings` function.